experimental/devices/mfrc522/mfrc522.go

```diff
@@ -176,66 +181,187 @@ func (r *Dev) Halt() error {
 //
 //	timeout the duration to wait for IRQ strobe.
 func (r *Dev) SetOperationTimeout(timeout time.Duration) {
+	r.oMu.Lock()
+	defer r.oMu.Unlock()
 	r.operationTimeout = timeout
 }
 
-// Init initializes the RFID chip.
-func (r *Dev) Init() error {
-	if err := r.Reset(); err != nil {
-		return err
+// SetAntennaGain configures antenna signal strength.
+//
+//	gain - signal strength from 0 to 7.
+func (r *Dev) SetAntennaGain(gain int) {
+	r.aMu.Lock()
+	defer r.aMu.Unlock()
+
+	if 0 <= gain && gain <= 7 {
```

Comment:
You can skip the mutex if the value is out of range. Preferably an error would be returned instead of a silent failure. So:
```
func (r *Dev) SetAntennaGain(gain int) error {
  if gain < 0 || gain > 7 {
    return errors.New("mfrc522: invalid gain")
  }
  r.aMu.Lock()
  r.antennaGain = gain
  r.aMu.Unlock()
  return nil
}

```